op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,278 @@
+		}
+	}
+}
+
+// startRollupBoostListener initializes and starts a WebSocket client to listen for rollup boost messages
+func (h *Handler) startRollupBoostListener(ctx context.Context) error {
+	if h.cfg.RollupBoostWsURL == "" {
+		h.log.Info("rollup boost WebSocket disabled, no URL configured")
+		return nil
+	}
+
+	// Start a goroutine to maintain the rollup boost WebSocket connection
+	go func() {
+		backoff := time.Second
+		maxBackoff := 30 * time.Second
+
+		for {
+			select {
+			case <-ctx.Done():
+				if h.rollupBoostConn != nil {
+					h.log.Info("closing rollup boost WebSocket connection due to context cancellation")
+					h.rollupBoostConn.Close()
+					h.rollupBoostConn = nil
+				}
+				return
+			default:
+				// Try to connect if not connected
+				if h.rollupBoostConn == nil {
+					conn, _, err := websocket.DefaultDialer.Dial(h.cfg.RollupBoostWsURL, nil)
+					if err != nil {
+						h.log.Error("failed to connect to rollup boost WebSocket, will retry", "err", err, "backoff", backoff)
+						time.Sleep(backoff)
+						// After each failed connection attempt, increases the wait time by multiplying by 1.5
+						// Caps the maximum wait time at 30 seconds
+						backoff = time.Duration(min(float64(backoff)*1.5, float64(maxBackoff)))
+						continue
+					}
+					h.rollupBoostConn = conn
+					h.log.Info("connected to rollup boost WebSocket", "url", h.cfg.RollupBoostWsURL)
+					backoff = time.Second // Reset backoff on successful connection
+				}
+
+				// Read messages from the connection
+				_, message, err := h.rollupBoostConn.ReadMessage()
+				if err != nil {
+					h.log.Error("error reading from rollup boost WebSocket, reconnecting", "err", err)
+					h.rollupBoostConn.Close()
+					h.rollupBoostConn = nil
+					continue
+				}
+
+				h.handleRollupBoostMessage(ctx, message)
+			}
+		}
+	}()
+
+	return nil
+}
+
+// handleRollupBoostMessage processes a message received from rollup boost
+func (h *Handler) handleRollupBoostMessage(ctx context.Context, message []byte) {
+	// Only forward messages if we're the leader - check dynamically each time
+	if !h.isLeaderFn(ctx) {
+		h.log.Debug("not forwarding rollup boost message, not the leader")
+		return
+	}
+
+	// Forward the message to connected clients
+	h.BroadcastMessage(message)
+}
+
+// handleWebSocketRequest processes new WebSocket connection requests
+func (h *Handler) handleWebSocketRequest(w http.ResponseWriter, r *http.Request, upgrader websocket.Upgrader, ctx context.Context) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		h.log.Error("failed to upgrade connection", "err", err)
+		return
+	}
+
+	// Store the client connection
+	h.wsClientMu.Lock()
+	// Add the new connection to our list of clients
+	h.wsClients = append(h.wsClients, conn)
+	h.wsClientMu.Unlock()
+
+	h.log.Info("WebSocket proxy connected", "remote", conn.RemoteAddr())
+
+	// Handle the connection in a separate goroutine
+	go h.handleWebSocketConnection(ctx, conn)
```

Comment:
```suggestion
	h.handleWebSocketConnection(ctx, conn)
```
this needs to be blocking so that the connection is actually preserved